### DISCUSSION TITLE
LSP tells me I am using a fallback Scala version. How do I set the version to the one it wants?

### DISCUSSION THREAD

**Discussion by mackler** [scalameta/metals#6775] [Q&A]
Hi, when I start using metals in emacs, I see this message:

`LSP :: You are using fallback Scala version 3.2.0, which is not supported in this version of Metals. Please upgrade to Scala version 3.3.3.`

In my `build.sbt` file I have the following line:

`ThisBuild / scalaVersion := "3.3.3"`

I installed Scala using Coursier.  When I run `scala -version` I get:

```
$ scala -version
Scala code runner version 3.3.3 -- Copyright 2002-2024, LAMP/EPFL
```

and the same for `scalac -version`:

```
$ scalac -version
Scala compiler version 3.3.3 -- Copyright 2002-2024, LAMP/EPFL
```

What am I doing to choose a version of Scala other than the one that's in the `build.sbt` file and that I installed using Coursier?  How can I "upgrade to Scala version 3.3.3" as the message is asking me to do?

Thank you.

**Comment by tgodzik** [comment]
I think that metals fallback version setting is set to 3.2.0 by default in some plugin (don't remember which one), while it should be "automatic"

The fallback version sometimes hows up before indexing or build import is finished, but here we have the added warning of a version we no longer release fixes for.

**Reply by mackler** [reply]
So is the message accurate that 3.2.0 is being used or is it just a result of the message preceding the build import?  Is there something I should do to use a supported version and/or avoid the message?

**Reply by tgodzik** [reply]
This should have been fixed actually https://github.com/emacs-lsp/lsp-metals/commit/0dc938be1190d147e7013e3dce08ac8bff5d1662

You might get information about a fallback version sometimes, but that's usually in scenarios I explained.
